src/shared/gpt.c

```diff
@@ -43,26 +187,31 @@ const GptPartitionType gpt_partition_type_table[] = {
         _GPT_ARCH_SEXTET(X86,         "x86"),
         _GPT_ARCH_SEXTET(X86_64,      "x86-64"),
 #ifdef SD_GPT_ROOT_NATIVE
-        { SD_GPT_ROOT_NATIVE,            "root",            native_architecture(), .is_root = true            },
-        { SD_GPT_ROOT_NATIVE_VERITY,     "root-verity",     native_architecture(), .is_root_verity = true     },
-        { SD_GPT_ROOT_NATIVE_VERITY_SIG, "root-verity-sig", native_architecture(), .is_root_verity_sig = true },
-        { SD_GPT_USR_NATIVE,             "usr",             native_architecture(), .is_usr = true             },
-        { SD_GPT_USR_NATIVE_VERITY,      "usr-verity",      native_architecture(), .is_usr_verity = true      },
-        { SD_GPT_USR_NATIVE_VERITY_SIG,  "usr-verity-sig",  native_architecture(), .is_usr_verity_sig = true  },
+        { SD_GPT_ROOT_NATIVE,            "root",            native_architecture(), .designator = PARTITION_ROOT            },
+        { SD_GPT_ROOT_NATIVE_VERITY,     "root-verity",     native_architecture(), .designator = PARTITION_ROOT_VERITY     },
+        { SD_GPT_ROOT_NATIVE_VERITY_SIG, "root-verity-sig", native_architecture(), .designator = PARTITION_ROOT_VERITY_SIG },
+        { SD_GPT_USR_NATIVE,             "usr",             native_architecture(), .designator = PARTITION_USR             },
+        { SD_GPT_USR_NATIVE_VERITY,      "usr-verity",      native_architecture(), .designator = PARTITION_USR_VERITY      },
+        { SD_GPT_USR_NATIVE_VERITY_SIG,  "usr-verity-sig",  native_architecture(), .designator = PARTITION_USR_VERITY_SIG  },
```

Comment:
I'm not sure if that matters, but now we'll have e.g.
```
{ SD_GPT_ROOT_ARM64,  "root-arm64", ARCHITECTURE_ARM64, .designator = PARTITION_ROOT_OTHER },
{ SD_GPT_ROOT_NATIVE,   "root",            native_architecture(),          .designator = PARTITION_ROOT            },
```
Which means that on arm64, the root partition in the first definition is labeled as `PARTITION_ROOT_OTHER`. This could be confusing… Maybe "OTHER" should be changed to "ARCH" ?

Reply:
I added descriptions to tell what these actually mean. I don't think changing the name is going to help much. We should probably get rid of the secondary/other enum variants altogether, keep track of all architectures while parsing the partition table, and just at the end figure out the ones we actually want in preference of primary architecture, then secondary, then any other, but let's leave this for another PR